### ISSUE TITLE
Non-existing layers shown in summary

### ISSUE THREAD

**Issue by lukasugar** [TylerYep/torchinfo#143]
**Describe the bug**
Summary shows some layers that don't exist.

**To Reproduce**
```
# Main model contains backbone and multiple heads

#init
self.backbone = timm.create_model('resnet50',...)

self.heads = torch.nn.ModuleList([SingleClassificationHead(
            num_input_channels=2048, activation='sigmoid') for i in range(5])

#forward
        y = self.backbone(x)

        outputs = []

        for head in self.heads:
            p = head(y)
            outputs.append(p)

        return outputs

# Head code:
class SingleClassificationHead(nn.Module):
    def __init__(self, num_input_channels):
        super(SingleClassificationHead, self).__init__()

        self.layers = []

        self.layers.append(nn.Conv2d(num_input_channels, 128, kernel_size=1))
        self.layers.append(nn.ReLU())
        self.layers.append(nn.Conv2d(128, 64, kernel_size=1))
        self.layers.append(nn.ReLU())
        self.layers.append(nn.Conv2d(64, 32, kernel_size=1))
        self.layers.append(nn.ReLU())
        self.layers.append(nn.Conv2d(32, 8, kernel_size=1))
        self.layers.append(nn.ReLU())
        self.layers.append(nn.Conv2d(8, 1, kernel_size=1))
        self.layers.append(nn.Sigmoid())

        self.layers = nn.Sequential(*self.layers)

    def forward(self, x):
        return self.layers(x)
```

When I run `summary(SingleClassificationHead(num_input_channels=2048), input_size=( 2048, 8, 8))`, the summary is looking good:
```
SingleClassificationHead(
  (layers): Sequential(
    (0): Conv2d(2048, 128, kernel_size=(1, 1), stride=(1, 1))
    (1): ReLU()
    (2): Conv2d(128, 64, kernel_size=(1, 1), stride=(1, 1))
    (3): ReLU()
    (4): Conv2d(64, 32, kernel_size=(1, 1), stride=(1, 1))
    (5): ReLU()
    (6): Conv2d(32, 8, kernel_size=(1, 1), stride=(1, 1))
    (7): ReLU()
    (8): Conv2d(8, 1, kernel_size=(1, 1), stride=(1, 1))
    (9): Sigmoid()
  )
)
----------------------------------------------------------------
        Layer (type)               Output Shape         Param #
================================================================
            Conv2d-1            [-1, 128, 8, 8]         262,272
              ReLU-2            [-1, 128, 8, 8]               0
            Conv2d-3             [-1, 64, 8, 8]           8,256
              ReLU-4             [-1, 64, 8, 8]               0
            Conv2d-5             [-1, 32, 8, 8]           2,080
              ReLU-6             [-1, 32, 8, 8]               0
            Conv2d-7              [-1, 8, 8, 8]             264
              ReLU-8              [-1, 8, 8, 8]               0
            Conv2d-9              [-1, 1, 8, 8]               9
          Sigmoid-10              [-1, 1, 8, 8]               0
================================================================
Total params: 272,881
Trainable params: 272,881
Non-trainable params: 0
----------------------------------------------------------------
Input size (MB): 0.50
Forward/backward pass size (MB): 0.23
Params size (MB): 1.04
Estimated Total Size (MB): 1.77
----------------------------------------------------------------
```
When i run `summary(model, input_size=(3, 128, 128))`, model creates additional sigmoids in each head. There should be one sigmoid per head, but there are _Nh_ sigmoids in each head, where _Nh_ is the number of heads:
```
#Printing out only the last part of backbone, and the heads
     BatchNorm2d-170           [-1, 2048, 4, 4]           4,096
            ReLU-171           [-1, 2048, 4, 4]               0
      Bottleneck-172           [-1, 2048, 4, 4]               0
        Identity-173           [-1, 2048, 4, 4]               0
        Identity-174           [-1, 2048, 4, 4]               0
SelectAdaptivePool2d-175           [-1, 2048, 4, 4]               0
        Identity-176           [-1, 2048, 4, 4]               0
          ResNet-177           [-1, 2048, 4, 4]               0
       AvgPool2d-178           [-1, 2048, 1, 1]               0
          Conv2d-179            [-1, 128, 1, 1]         262,272
            ReLU-180            [-1, 128, 1, 1]               0
          Conv2d-181             [-1, 64, 1, 1]           8,256
            ReLU-182             [-1, 64, 1, 1]               0
          Conv2d-183             [-1, 32, 1, 1]           2,080
            ReLU-184             [-1, 32, 1, 1]               0
          Conv2d-185              [-1, 8, 1, 1]             264
            ReLU-186              [-1, 8, 1, 1]               0
          Conv2d-187              [-1, 1, 1, 1]               9
         Sigmoid-188              [-1, 1, 1, 1]               0
         Sigmoid-189              [-1, 1, 1, 1]               0
         Sigmoid-190              [-1, 1, 1, 1]               0
         Sigmoid-191              [-1, 1, 1, 1]               0
         Sigmoid-192              [-1, 1, 1, 1]               0
SingleClassificationHead-193              [-1, 1, 1, 1]               0
          Conv2d-194            [-1, 128, 1, 1]         262,272
            ReLU-195            [-1, 128, 1, 1]               0
          Conv2d-196             [-1, 64, 1, 1]           8,256
            ReLU-197             [-1, 64, 1, 1]               0
          Conv2d-198             [-1, 32, 1, 1]           2,080
            ReLU-199             [-1, 32, 1, 1]               0
          Conv2d-200              [-1, 8, 1, 1]             264
            ReLU-201              [-1, 8, 1, 1]               0
          Conv2d-202              [-1, 1, 1, 1]               9
         Sigmoid-203              [-1, 1, 1, 1]               0
         Sigmoid-204              [-1, 1, 1, 1]               0
         Sigmoid-205              [-1, 1, 1, 1]               0
         Sigmoid-206              [-1, 1, 1, 1]               0
         Sigmoid-207              [-1, 1, 1, 1]               0
SingleClassificationHead-208              [-1, 1, 1, 1]               0
          Conv2d-209            [-1, 128, 1, 1]         262,272
            ReLU-210            [-1, 128, 1, 1]               0
          Conv2d-211             [-1, 64, 1, 1]           8,256
            ReLU-212             [-1, 64, 1, 1]               0
          Conv2d-213             [-1, 32, 1, 1]           2,080
            ReLU-214             [-1, 32, 1, 1]               0
          Conv2d-215              [-1, 8, 1, 1]             264
            ReLU-216              [-1, 8, 1, 1]               0
          Conv2d-217              [-1, 1, 1, 1]               9
         Sigmoid-218              [-1, 1, 1, 1]               0
         Sigmoid-219              [-1, 1, 1, 1]               0
         Sigmoid-220              [-1, 1, 1, 1]               0
         Sigmoid-221              [-1, 1, 1, 1]               0
         Sigmoid-222              [-1, 1, 1, 1]               0
SingleClassificationHead-223              [-1, 1, 1, 1]               0
          Conv2d-224            [-1, 128, 1, 1]         262,272
            ReLU-225            [-1, 128, 1, 1]               0
          Conv2d-226             [-1, 64, 1, 1]           8,256
            ReLU-227             [-1, 64, 1, 1]               0
          Conv2d-228             [-1, 32, 1, 1]           2,080
            ReLU-229             [-1, 32, 1, 1]               0
          Conv2d-230              [-1, 8, 1, 1]             264
            ReLU-231              [-1, 8, 1, 1]               0
          Conv2d-232              [-1, 1, 1, 1]               9
         Sigmoid-233              [-1, 1, 1, 1]               0
         Sigmoid-234              [-1, 1, 1, 1]               0
         Sigmoid-235              [-1, 1, 1, 1]               0
         Sigmoid-236              [-1, 1, 1, 1]               0
         Sigmoid-237              [-1, 1, 1, 1]               0
SingleClassificationHead-238              [-1, 1, 1, 1]               0
          Conv2d-239            [-1, 128, 1, 1]         262,272
            ReLU-240            [-1, 128, 1, 1]               0
          Conv2d-241             [-1, 64, 1, 1]           8,256
            ReLU-242             [-1, 64, 1, 1]               0
          Conv2d-243             [-1, 32, 1, 1]           2,080
            ReLU-244             [-1, 32, 1, 1]               0
          Conv2d-245              [-1, 8, 1, 1]             264
            ReLU-246              [-1, 8, 1, 1]               0
          Conv2d-247              [-1, 1, 1, 1]               9
         Sigmoid-248              [-1, 1, 1, 1]               0
         Sigmoid-249              [-1, 1, 1, 1]               0
         Sigmoid-250              [-1, 1, 1, 1]               0
         Sigmoid-251              [-1, 1, 1, 1]               0
         Sigmoid-252              [-1, 1, 1, 1]               0
SingleClassificationHead-253              [-1, 1, 1, 1]               0
================================================================
Total params: 24,872,437
Trainable params: 24,872,437
Non-trainable params: 0
----------------------------------------------------------------
Input size (MB): 0.19
Forward/backward pass size (MB): 94.85
Params size (MB): 94.88
Estimated Total Size (MB): 189.91
----------------------------------------------------------------
```

To verify that the network is built properly, and that the additional sigmoids don't exist, I've run `print(model)`:
```
# showing only the last part of print, containing heads:
        (7): ReLU()
        (8): Conv2d(8, 1, kernel_size=(1, 1), stride=(1, 1))
        (9): Sigmoid()
      )
    )
    (3): SingleClassificationHead(
      (layers): Sequential(
        (0): Conv2d(2048, 128, kernel_size=(1, 1), stride=(1, 1))
        (1): ReLU()
        (2): Conv2d(128, 64, kernel_size=(1, 1), stride=(1, 1))
        (3): ReLU()
        (4): Conv2d(64, 32, kernel_size=(1, 1), stride=(1, 1))
        (5): ReLU()
        (6): Conv2d(32, 8, kernel_size=(1, 1), stride=(1, 1))
        (7): ReLU()
        (8): Conv2d(8, 1, kernel_size=(1, 1), stride=(1, 1))
        (9): Sigmoid()
      )
    )
    (4): SingleClassificationHead(
      (layers): Sequential(
        (0): Conv2d(2048, 128, kernel_size=(1, 1), stride=(1, 1))
        (1): ReLU()
        (2): Conv2d(128, 64, kernel_size=(1, 1), stride=(1, 1))
        (3): ReLU()
        (4): Conv2d(64, 32, kernel_size=(1, 1), stride=(1, 1))
        (5): ReLU()
        (6): Conv2d(32, 8, kernel_size=(1, 1), stride=(1, 1))
        (7): ReLU()
        (8): Conv2d(8, 1, kernel_size=(1, 1), stride=(1, 1))
        (9): Sigmoid()
      )
    )
  )
```

**Expected behavior**
It's expected that summary prints only one sigmoid per head

**Screenshots**

**Desktop (please complete the following information):**
 - OS: Windows 11, but using WSL2

**Additional context**
Using:
torchsummary 1.5.1
torch 1.11.0
pytorch-lightning 1.6.1

**Comment by TylerYep** [comment]
Please install torchinfo instead. If you still see this issue, then modify the output to show what torchinfo outputs.
